### PR TITLE
Allow rpc changes requests to fetch ddoc from local cache

### PR DESCRIPTION
**Pull request by chewbranca**

### CODE REVIEW

**Comment by chewbranca** [comment]
Companion PR out in https://github.com/cloudant/dbcore/pull/265.

File: src/fabric_rpc2.erl [hunk 1]
```diff
@@ -96,7 +96,14 @@ changes(DbName, #changes_args{} = Args, StartVector, DbOptions) ->
     changes(DbName, [Args], StartVector, DbOptions);
 changes(DbName, Options, StartVector, DbOptions) ->
     set_io_priority(DbName, DbOptions),
-    #changes_args{dir=Dir} = Args = lists:keyfind(changes_args, 1, Options),
+    #changes_args{dir=Dir, filter=Filter} = Args0 = lists:keyfind(changes_args, 1, Options),
+    Args = case Filter of
+        {fetch, Style, {DbArg, JsonReq, DDocId, FName}} ->
```
Comment (davisp) [inline src/fabric_rpc2.erl]: Seems like you need a `when is_binary(DDocId)` here.

Comment (chewbranca) [inline src/fabric_rpc2.erl]: I suppose it wouldn't hurt to add that in, but right now the only caller sends a binary. Do you think it's better to fail loudly with a pattern match error here, or silently skip the fetch command and load the slow way? I think I would honestly rather see this throw a 500 than fail silently back to the slow approach.

Comment (davisp) [inline src/fabric_rpc2.erl]: Also I think we should move this to couch_changes as an API function. I don't like introspecting that filter tuple outside of the module that defines it.

**Comment by davisp** [comment]
+1

**Pull request merged**